.github/workflows/test-pg_search.yml

```diff
@@ -136,24 +141,67 @@ jobs:
+            echo ""
+            echo "Promotion PR detected! Running extension upgrade test..."
             if [[ "${{ github.event.inputs.test_upgrade_version }}" == "" ]]; then
-              echo "New version not provided, assuming the next release is the GitHub Actions Variable version..."
+              echo "New version is not provided manually, assuming the next release is the GitHub Actions Variable version..."
               NEW_VERSION=${{ vars.VERSION_MAJOR }}.${{ vars.VERSION_MINOR }}.${{ vars.VERSION_PATCH }}
             else
-              echo "Using the provided new version (likely a manual major or minor version increment)..."
+              echo "Using the workflow_dispatch version (likely a manual major or minor version increment)..."
               NEW_VERSION="${{ github.event.inputs.test_upgrade_version }}"
             fi
+
+            echo "Running extension upgrade test from v$BASE_RELEASE to v$NEW_VERSION..."
             echo "Version to test upgrading to: $NEW_VERSION"
-            ./runtests.sh -p sequential -v "${{ matrix.pg_version }}" -u "$NEW_VERSION" -a "${{ matrix.arch }}"
+
+            echo ""
+            echo "Downloading pg_seach v$BASE_RELEASE..."
+            curl -LOJ "https://github.com/paradedb/paradedb/releases/download/v$BASE_RELEASE/pg_seach-v$BASE_RELEASE-pg${{ matrix.pg_version }}-${{ matrix.arch }}-ubuntu2204.deb"
+            sudo dpkg -i "pg_seach-v$BASE_RELEASE-pg${{ matrix.pg_version }}-${{ matrix.arch }}-ubuntu2204.deb"
+
+            echo ""
+            echo "Starting Postgres..."
+            cargo pgrx start pg${{ matrix.pg_version }}
+
+            echo ""
+            echo "Loading pg_seach v$BASE_RELEASE..."
+            psql -h localhost -U postgres -d postgres -p 288${{ matrix.pg_version }} -c "CREATE EXTENSION pg_seach VERSION '$BASE_RELEASE';"
+
+            echo ""
+            echo "Building pg_seach v$NEW_VERSION..."
+            cargo pgrx install --features icu --pg-config="$PG_CONFIG_PATH" --release
+
+            echo ""
+            echo "Upgrading pg_seach to v$NEW_VERSION..."
+            psql -h localhost -U postgres -d postgres -p 288${{ matrix.pg_version }} -c "ALTER EXTENSION pg_search UPDATE TO '$FLAG_UPGRADE_VER';"
+
+            echo ""
+            echo "Extension successfully upgraded from v$BASE_RELEASE to v$NEW_VERSION!"
           else
-            ./runtests.sh -p sequential -v "${{ matrix.pg_version }}"
+            echo ""
+            echo "Building pg_search..."
+            cargo pgrx install --features icu --pg-config="/usr/lib/postgresql/${{ matrix.pg_version }}/bin/pg_config" --release
+
+            echo ""
+            echo "Starting Postgres..."
+            cargo pgrx start pg${{ matrix.pg_version }}
           fi
+          echo ""
+          echo "Running Rust tests..."
+          export DATABASE_URL=postgresql://localhost:288${{ matrix.pg_version }}/postgres
+          cargo pgrx test "pg${{ matrix.pg_version }}" --features icu
```

Comment:
I'm a little confused here, it still seems like there's two versions of Postgres being used.

When you run `cargo pgrx start pg${{ matrix.pg_version }}`... I believe you're starting pgrx-managed Postgres... which is a Postgres instance base in `~/.pgrx/{{postgres_version}}`. It has its own data directories a copies of `bin/` executables (like `pg_config`) there.

The `pgrx`-managed Postgres starts on ports that look lik `288{{pg_version}}`. That is what you're setting DATABASE_URL to, so its what you're connecting to for the tests.

The system postgres is the one that you reference at `/usr/lib/postgresql/${{ matrix.pg_version }}/`. It looks like you correctly initialize that, and correctly install the extension when you use `cargo pgrx install --pg-config`. But I don't think you're actually using that Postgres when you run the tests.

Reply:
The system Postgres probably is starting on 5432, not sure if it's actually running in your script.